make/tests

```diff
@@ -47,7 +47,19 @@ while has_pod ; do
     sleep 1
 done
 
-kubectl create --namespace="${NAMESPACE}" --filename="output/kube/bosh-task/${POD_NAME}.yaml"
+# Replace the fixed secret in the relevant task definition with the
+# actual name as pulled from the cluster under test.
+
+capsecret="$(kubectl get pod api-0 --namespace "${NAMESPACE}" -o jsonpath='{@.spec.containers[0].env[?(@.name=="MONIT_PASSWORD")].valueFrom.secretKeyRef.name}')"
+#echo cap-secret = $capsecret
+
+CONFIG=$(mktemp)
+sed < "output/kube/bosh-task/${POD_NAME}.yaml" \
+    > "${CONFIG}" \
+    "s|name: \"secret\"|name: \"${capsecret}\"|"
+
+kubectl create --namespace="${NAMESPACE}" --filename="${CONFIG}"
+rm "${CONFIG}"
```

Comment:
We have `errexit` in effect here; would that not have a chance to not delete the temp file?

Reply:
Oh, so you think we still need to trap ?

Reply:
Re-added with commit a48233d7d1a27f867a07a4cd138bd4248e05cd43